### PR TITLE
Add checks to the dangerfile to fail the build on certain style issues, in lieu of prettier

### PR DESCRIPTION
**Pull request by weswigham**
I was pinged by activity on a certain DT PR which reminded me we hadn't implemented hard checks for this yet (I don't think). Hopefully this should end some discussions in PRs, at least (and ideally move it to a separate thread, so those PRs can go on with their changes according to our current style settings),

Naturally a DT maintainer could always merge something even in the face of these failures, but what's important is that these checks prevent any potential automerges with them.

### CODE REVIEW

File: dangerfile.ts [hunk 1]
```diff
@@ -53,3 +54,15 @@ if (fs.existsSync(suggestionsDir)) {
...
+    fail("A nested .editorconfig file may not be added to a package on DefinitelyTyped. Please respect the root .editorconfig.", danger.git.created_files.find(f => path.basename(f) === ".editorconfig"), 1)
+}
+
+
+for (const filename of danger.git.modified_files.concat(danger.git.created_files)) {
+    danger.git.diffForFile(filename).then(d => {
+        if (d.added.indexOf("\t") > -1) { // This is a dumb check for tabs, in lieu of running `prettier` on the diff'd parts
+            fail("The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.", filename, d.after.split("\n").findIndex(e => e.indexOf("\t") > -1) + 1)
+        }
+    })
```
Comment (weswigham) [inline dangerfile.ts:67]: @orta I'm nor familiar with async code in a dangerfile - do I need to hook up a `.catch` on these callbacks, or do I not? Is it fine as-is? Would we prefer an `async` function we call (and then do we need to `.catch` on that)?

Comment (orta) [reply dangerfile.ts:67]: Nah, this is fine - Danger's execution path is roughly that when the synchronous evaluation runs, then it hooks into the process exit code to catch all loose promises

**Comment by orta** [comment]
If folks are interested in knowing how to test something like this:

Failing:
```sh
❯ yarn danger pr https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42427
yarn run v1.22.4
$ /Users/ortatherox/dev/typescript/DefinitelyTyped/node_modules/.bin/danger pr https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42427
Starting Danger PR on DefinitelyTyped/DefinitelyTyped#42427

Danger: ⅹ Failing the build, there is 1 fail.
## Failures
A nested .editorconfig file may not be added to a package on DefinitelyTyped. Please respect the root .editorconfig.
✨  Done in 6.04s.
```

Passing:
```sh
❯ yarn danger pr https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44662
yarn run v1.22.4
$ /Users/ortatherox/dev/typescript/DefinitelyTyped/node_modules/.bin/danger pr https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44662
Starting Danger PR on DefinitelyTyped/DefinitelyTyped#44662

Danger: ✓ passed review, received no feedback.
```

**Review by orta — APPROVED** [review]

File: dangerfile.ts (hunk 1 above)
Comment (orta) [reply dangerfile.ts:67]: Oh yeah, the original point was about catching - Danger has a root catch which will fail the build and post a message with the stack, IMO, this shouldn't ever catch but it's handled at runtime level

**Review by sandersn — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**